src/bscript.js

```diff
@@ -1,5 +1,31 @@
+const bs58check = require('bs58check')
+const bech32lib = require('bech32')
+
 const parseScript = require('./parseScript')
 const parseAsm = require('./parseAsm')
+const {
+  matchScriptType,
+  executeTemplate,
+  p2sh: p2shTemplate,
+  p2pkh: p2pkhTemplate
+} = require('./templates')
+
+function byteToBuffer (byte) {
+  const buf = Buffer.alloc(1)
+  buf.writeUInt8(byte)
+  return buf
+}
+
+const ADDRESS_SCRIPT_TYPES = [
+  'p2pkh',
+  'p2sh',
+  'p2wsh',
+  'p2wpkh'
+]
+
+const PUBKEY_HASH = 0x00
```

Comment:
I know this is for bitcoin, but it would be good to parameterize these to make this library available to other similar coins.

Reply:
@JBaczuk These constants are just defaults.  They can be passed in as the `pubKeyHash`, `scriptHash`, and `bech32` options.